concrete/src/Routing/RoutingServiceProvider.php

```diff
@@ -10,7 +10,7 @@ class RoutingServiceProvider extends Provider
      */
     public function register()
     {
-        $this->app->singleton(RouterInterface::class);
+        $this->app->singleton(Router::class);
         $this->app->bind(RouterInterface::class, Router::class);
```

Comment:
That's a good point, we can probably be using aliases more often. Your example doesn't really solve the problem though, imagine if you first rebound `RouterInterface::class` to `CustomRouter::class` and then ran your example test replacing "Router" with "CustomRouter". Are you sure your singleton binding would carry over?